### ISSUE TITLE
No IMPLICIT type errors in dispdftd3.f90

### ISSUE THREAD

**Issue by lexming** [Atomistica/atomistica#27]
Compilation of Atomistica v0.10.1 for ASE following the instructions in [doc/install.python.md](https://github.com/Atomistica/atomistica/blob/master/doc/install.python.md) fails with `no IMPLICIT type` errors due to missing type declarations in `dispdftd3.f90`
```
/theia/scratch/brussel/101/vsc10122/easybuild-ivb/build/Atomistica/0.10.1/foss-2019b-Python-3.7.4-ASE/atomistica-0.10.1/src/potentials/dispersion/dispdftd3.f90:266:31:

     if (system_of_units == eV_A .or. system_of_units == eV_A_fs) then
                               1
Error: Symbol 'ev_a' at (1) has no IMPLICIT type
/theia/scratch/brussel/101/vsc10122/easybuild-ivb/build/Atomistica/0.10.1/foss-2019b-Python-3.7.4-ASE/atomistica-0.10.1/src/potentials/dispersion/dispdftd3.f90:266:63:

     if (system_of_units == eV_A .or. system_of_units == eV_A_fs) then
                                                               1
Error: Symbol 'ev_a_fs' at (1) has no IMPLICIT type
/theia/scratch/brussel/101/vsc10122/easybuild-ivb/build/Atomistica/0.10.1/foss-2019b-Python-3.7.4-ASE/atomistica-0.10.1/src/potentials/dispersion/dispdftd3.f90:261:82:

     coords(1:3, 1:p%nat) = POS3(p, 1:p%nat) * length_to_Bohr
                                                                                  1
Error: Symbol 'length_to_bohr' at (1) has no IMPLICIT type
/theia/scratch/brussel/101/vsc10122/easybuild-ivb/build/Atomistica/0.10.1/foss-2019b-Python-3.7.4-ASE/atomistica-0.10.1/src/potentials/dispersion/dispdftd3.f90:266:23:

     if (system_of_units == eV_A .or. system_of_units == eV_A_fs) then
                       1
Error: Symbol 'system_of_units' at (1) has no IMPLICIT type
/theia/scratch/brussel/101/vsc10122/easybuild-ivb/build/Atomistica/0.10.1/foss-2019b-Python-3.7.4-ASE/atomistica-0.10.1/src/potentials/dispersion/dispdftd3.f90:138:47:

     input%cutoff = this%cutoff * length_to_Bohr
                                               1
Error: Symbol 'length_to_bohr' at (1) has no IMPLICIT type
```

This is the specific compilation command

>error: Command "gfortran -cpp -fPIC -ffree-form -ffree-line-length-none -x f95-cpp-input -DHAVE_DFTD3 -I/user/brussel/101/vsc10122/.local/easybuild/software/dftd3-lib/0.9-GCC-8.3.0/include -DHAVE_FFTW3 -I/apps/brussel/CO7/ivybridge-ib/software/FFTW/3.3.8-gompi-2019b/include -fopenmp -O2 -ftree-vectorize -march=native -fno-math-errno -fPIC -O3 -funroll-loops -DNO_BIND_C_OPTIONAL -DPYTHON -I/apps/brussel/CO7/ivybridge-ib/software/SciPy-bundle/2019.10-foss-2019b-Python-3.7.4/lib/python3.7/site-packages/numpy/core/include -Ibuild -Isrc -Isrc/support -Isrc/potentials -Isrc/notb -Isrc/notb/dense -I/apps/brussel/CO7/ivybridge-ib/software/SciPy-bundle/2019.10-foss-2019b-Python-3.7.4/lib/python3.7/site-packages/numpy/core/include -c -c /theia/scratch/brussel/101/vsc10122/easybuild-ivb/build/Atomistica/0.10.1/foss-2019b-Python-3.7.4-ASE/atomistica-0.10.1/src/potentials/dispersion/dispdftd3.f90 -o build/temp.linux-x86_64-3.7/theia/scratch/brussel/101/vsc10122/easybuild-ivb/build/Atomistica/0.10.1/foss-2019b-Python-3.7.4-ASE/atomistica-0.10.1/src/potentials/dispersion/dispdftd3.o -Jbuild/temp.linux-x86_64-3.7 -Ibuild/temp.linux-x86_64-3.7" failed with exit status 1

I solved it by adding the appropriate type definitions in `dispdftd3.f90` from `src/standalone/particles.f90`, but I'm not sure if this is the correct solution or something else is wrong/missing in `setup.py`.

**Comment by pastewka** [comment]
Did you add -DHAVE_DFTD3 manually? This is experimental for the Python interface at present. I'll fix this. The problem is that system_of_units is defined only for the standalone interface, LAMMPS and PYTHON is always eV/A.

**Comment by pastewka** [comment]
I just pushed a fix to master. Can you check whether this fixes this issue for you?

Does DFD-D3 work within Python for you? I think there may still be something else that is broken. The standalone code should be okay.

**Comment by lexming** [comment]
Thanks for the quick fix. The patch in https://github.com/Atomistica/atomistica/commit/6a26681600bf433fd22ca679127678554a3f212d does fix this issue and compilation completes.

To be clear, this issues is independent of `-DHAVE_DFTD3`. What I did is
* download the code for v0.10.1
* copy `setup.cfg.gnu_omp` to `setup.cfg`
* execute `python setup.py build`

With the last patch, the compilation of `dispdftd3.f90` proceeds without issues.

Regarding `-DHAVE_DFTD3`, I added it manually through `setup.cfg` once I saw that `dispdftd3.f90` had code for it. Mainly because we already have dftd3-lib installed. Can I run any specific test to check that DFTD3 is working as intended? or is it just safer to leave disabled for the Python interface of Atomistica?
